py/objproperty.c

```diff
@@ -38,28 +38,25 @@ typedef struct _mp_obj_property_t {
 } mp_obj_property_t;
 
 STATIC mp_obj_t property_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 4, false);
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_fget, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_fset, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_fdel, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_doc, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+    };
+    mp_arg_val_t vals[MP_ARRAY_SIZE(allowed_args)];
+    mp_map_t kw_args;
+    mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
+    mp_arg_parse_all(n_args, args, &kw_args, MP_ARRAY_SIZE(vals), allowed_args, vals);
```

Comment:
There is a new pattern for doing this that allows you to access the arguments by named struct field: https://github.com/micropython/micropython/blob/master/stmhal/uart.c#L460-L465